samples/smoketest/EventHubs.py

```diff
@@ -0,0 +1,52 @@
+from azure.eventhub import EventHubClient, EventData, EventPosition
+from datetime import datetime
+import os
+
+class EventHub:
+    def __init__(self):
+        # This test requires a previusly created Event Hub.
+        # In this example the name is "myeventhub", but it could be change below
+        connectionString = os.environ["EVENT_HUBS_CONNECTION_STRING"]
+        eventHubName = 'myeventhub'
+        self.client = EventHubClient.from_connection_string(connectionString, eventHubName)
+
+    def GetPartitionIds(self):
+        print("Getting partitions id...")
+        partition_ids = self.client.get_partition_ids()
+        print("\tdone")
+        return partition_ids      
+
+    def SendAndReceiveEvents(self, partitionID):
+        consumer = self.client.create_consumer(consumer_group="$default", partition_id=partitionID, event_position=EventPosition(datetime.utcnow()))
```

Comment:
`EventHubProducer` and `EventHubConsumer` are context managers, which means rather than calling their `close` methods, you can use them in more Pythonic `with` statements:

```py
with self.client.create_producer(partition_id=partitionID) as producer:
    # use producer as you like, it will be closed at the end of this block
```